### DISCUSSION TITLE
Wireless RS485 connection - EW11

### DISCUSSION THREAD

**Discussion by sfeakes** [aqualinkd/AqualinkD#336] [General]
There are a few discussions and people playing with EW-11 to connect to the Jandy control panel over wireless connection, so brining all the discussions into one place.

At the moment people have had good results with RS panels, and somewhat ok results with PDA panels. I personally think their will always be problems with PDA panels.

Few threads.
Below has most information, and screen shots of config.
https://github.com/sfeakes/AqualinkD/issues/300

Some more details on settings/config for EW11.  Note the GAP setting. 
https://github.com/sfeakes/AqualinkD/discussions/333

**Comment by bcre3306** [comment]
[EW11 - Communications Settings.pdf](https://github.com/user-attachments/files/16696482/EW11.-.Communications.Settings.pdf)
[EW11-Serial Port Settings.pdf](https://github.com/user-attachments/files/16696483/EW11-Serial.Port.Settings.pdf)
[EW11-System Settings.pdf](https://github.com/user-attachments/files/16696484/EW11-System.Settings.pdf)
![IMG_2521](https://github.com/user-attachments/assets/b0a6705d-b630-471a-91d0-b2f7f102af81)
Uploaded all documents here to help.  Also attached a picture of the wiring for the EW11. 

Hope this helps

**Reply by sts098** [reply]
bcre3306, seems you and I are working on much the same system pd8-combo.

Just checking.  You mentioned changing gap to 10 ms in ew11 per my post.  Did you also set:

rs485_frame_delay = 0

did this get rid of your bad packet warnings?  

On mine it greatly reduced the warnings, but I still have get on average about 1 per minute.  This is much better than the 6+ per minute I had in the past.

I am overloading the comm port, 4 connections as one of my ports is not working.  I am curious if you are having better results.  I plan to play with wiring to see if I can get some further improvements.  Do you have 2 rs485 ports?  Do both work?


Thanks!

**Reply by originalprime** [reply]
Hey @bcre3306 thanks for the leg work on this. My current setup uses the FTDI cable mentioned in the beginning of the installation instructions, but the problem is that I have chewed through several of those cables already. Nearby lightning strikes and other power surges just wreak havoc on the poor rs458 cabling. It's a miracle my Aqualink keypad still works.

I have a few questions about your setup. Where is the wifi antenna? Are you using the Elfin unit with the built-in antenna or are you using the one with the external antenna? How did you mount the Elfin? How did you mount the external antenna, if you're using one?

And have you tried any of the PoE / 485-to-ethernet adapters? By a stroke of luck, I actually have an ethernet run inside of my pool controller box, and I always prefer to be on the wire whenever possible. But if the Elfin units are known to work, that's pretty compelling, assuming I can figure out how to mount the damn thing and get the antenna outside of the metal box.

**Reply by bcre3306** [reply]
Hi @originalprime.  I actually gave up with it because I was having issues that ended up being related to it but let me answer as many of your questions as I can.

I bought both the Elfin EW11 with and without the antenna.  The one without the antenna dropped packets like crazy most likely because I had it mounted in the electrical panel for the RS4 panel.  When I got the unit with the antenna I barely had any communication issues but still had other issues which cause me to flip over to a PI instead.  With the unit with the external antenna I had the antenna poking out of the bottom of the electrical panel where a chase would normally be which allowed for good connection.  If I was going to keep it permanetly I was going to mount an outdoor antenna on the top of the box and run a cable back to the EW11 antenna connection.

I don't have ethernet at the panel, if I did then I would of eliminated wifi all together and hooked it up via a PI with ethernet.

Hope this helps

**Reply by sts098** [reply]
Thought I would jump in.  I am using the EW11 with the external antenna.  I had issues with some earlier versions of Aqualinkd, but am happy with current operation.  I am running a PDA board which, I believe, has increased serial traffic.  

Below is a photo:
![2024-10-26 18 12 05](https://github.com/user-attachments/assets/64b3313e-4e11-47ea-bd07-017112e265bd)
Doesn't really show much.  I have the EW11 in the enclosure and ran an extension cable with antenna outside the box.   

The EW11 has been up 27 days and 5 plus hours.  Wifi RSSI from Ew11 is 72 (so not great), but it seems to function well.   I get at most 1 or 2 checksum warnings in an hour, which doesn't seem to affect operation.  The ew11 lists 0 failed frames on serial port state and 0 failed frames on netp.

Just for info, I am running 2.5.0 dev 0.2.  on a PDA-PS8  Combo on version 7.1.0.  I am able to control from website and monitoring on Home Assistant.  Haven't tested HA control yet.  Aqualinkd is installed on a VM, not on a PI.  

I get a couple of errors during re-start:
Error:  RS Serial: Didn't find valid blocking serial port file descriptor
Error:  PDA:       PDA Init :- wait for next menu failed

And seem to generate this error when starting the Spa Blower
Error:  PDA:       PDA Device On/Off: Air Blower on - wait for CMD_MSG_LONG

Not sure if these error are issues with my configuration, or communications/timing, or something else.  

None of this seems to be affecting operation.  I only noticed when using aqmanager.html

I am very happy with the EW11.

**Reply by originalprime** [reply]
@bcre3306 Thank you for your response. Sorry to hear about your unsatisfactory experiences with the Elfin devices, but that is good information to have.

If I proceed, I think I would get the one with the external antenna and mount an antenna on the side or top of the pool equipment box. We actually have good wifi coverage throughout the yard so it should pick up.

I had actually thought about putting the Pi itself inside the pool equipment box but I thought that would cook it so I never tried it. It would make for a good experiment though, so I might just do that. Still not sure if it'll help the problem of the FTDI cables dying constantly, but it might.

I think I'm going to get an Elfin anyway and try that experiment.

**Reply by bcre3306** [reply]
My challenges with the EW11 were potentially related to my panel version or my wifi, never figured out exactly.  I was having ghosting issues where equipment would spastically turn on and off so it wasn't something I could leave running when I would leave the house.

**Reply by bcre3306** [reply]
@originalprime if you are going to get one, I would suggest you get it from aliexpress as they are half the price of getting it on amazon(at least when I was looking).

**Reply by originalprime** [reply]
@sts098 Thank you very much for your response. The magnet antenna holder is simplistic brilliance.

Do you have a link to the extension cable you bought for this project, and are you just running it out of one of the chase holes on the bottom or did you make a hole for that antenna extension?

I actually have good wifi coverage in the yard so I am not too worried about the wifi side of the equation. As I told @bcre3306 I think I'm going to buy an Elfin at this point just for the experiment, but if that fails I will probably end up shoving my Raspberry Pi inside the pool equipment box and hardwire it via ethernet and PoE, which I am fortunate enough to have at the pool equipment deck. The only reason I hadn't done this to start with is because I thought I would cook the Pi, and when I started this project a new Raspberry Pi was hard to come by. Further, I am not entirely convinced that moving the Pi to the inside of the pool equipment box will solve the problem I'm having, which is some kind of interference which keeps damaging my FTDI cables. But maybe it will because I will be connected directly to the bus instead of sharing a port with the wall keypad thing.

Another reason to try the Elfin is that it would allow me to free up the Pi for other projects. If I can connect to the Jandy RS485 bus over my network then I can slap AqualinkD in a container and call it a day. That might be a nice approach.

As for the errors you mentioned, I am not sure I would see something similar because I am using an RS8, so hopefully I wouldn't have to work around those same errors.

**Reply by sts098** [reply]
Bought this cable from Amazon.
CAPChang RG174 Coaxial Cable RP SMA Antenna Extension Cable 2-Pack 6.5FTRP-SMA Male to RP-SMA Female Bulkhead Mount Connectors Low Loss Antenna Cable for WiFi Antenna WiFi Router Security IP Camera

ran cable through one of the bottom holes.

**Reply by knjpollard** [reply]
I didn’t have exactly the same requirements, but I will share what I did to route my serial over IP wirelessly using raspberry pi. 

I created a python based serial server that listens on a specific port for requests from anyone. This server just reads from /dev/ttyUB0 which is receiving all chatter from an RS8 (rev Y). 

On the client side I have another Pi with AqualinkD running on it and it connected to the server. It reads the serial stream and worked successfully in AqualinkD as if it was running locally to the RSlink board.

**Comment by sts098** [comment]
replying based on thread 333.

The temperature issue I mentioned in earlier thread seem to do with:
convert_mqtt_temp_to_c = yes
convert_dz_temp_to_c = yes

my temps are in F.  104, and 70.  With these settings on, it is taking them as c and converting to f.  104 goes to 219.  70 to 158.  Then tried to update. 

Changing to no corrected the issue.

**Comment by bcre3306** [comment]
@sts098 sorry I have a different panel(RS4 Combo).  I haven't tried the frame delay setting but I am getting an error more then once a min but MUCH cleaner then it was.  I'll give that a try and report back in a little while. My temps are coming across correctly also

**Reply by sts098** [reply]
I get temps, pump rpm and watts.  I see changes to pump, cleaner, waterfall, spa and heater.  Most of these I can control.  Variable speed pump is a little hinky.  Can get out of sync on Mqtt and Aqualinkd.  Haven’t been able to repeatably isolate issue yet.

**Comment by bcre3306** [comment]
The only warning I really get now(minus packet drops which are because I need the EW11 with the external antenna which is on order) are the ones below. I get them about every 10 seconds


Warning:RS Serial: Read  Jandy   packet BAD PACKET To 0x33 of type   Unknown '0x72' | HEX: 0x10|0x02|0x33|0x72|0x20|0x01|0x02|0x03|0x04|0x05|0x06|0x07|0x08|0x09|0x0a|0x0b|0x0c|0x0d|0x0e|0

**Reply by sfeakes** [reply]
From looking at the log you posted in a different thread. I think that the bad packet you are seeing, is due to a protocol change in Board Rev Yg that has changed, and not an actual RS485 issue with the EW11.

I will look at posting a fix for this in the next release.

**Comment by sfeakes** [comment]
After some initial testing with different panels, I have found the following settings to work best for EW11.
For socat. (change IP 1.1.1.1 to you EW11's ip)
```
socat -d -d pty,link=/dev/ttyEW11,raw,ignoreeof TCP4:1.1.1.1:8899,ignoreeof
```
In AqualinkD the below are important (port matches above, no frame delay)
```
serial_port=/dev/ttyEW11
rs485_frame_delay = 0
```
On the EW11. Note the buffer and gap time are set to the minimum.
![Screenshot 2024-09-17 at 1 33 30 PM](https://github.com/user-attachments/assets/929bffe4-2e26-4fc1-b493-2f3fd04e7e80)

**Reply by jkosharek** [reply]
Running AquaLinkD on a Debian VM and can run this '`socat -d -d pty,link=/dev/ttyEW11,raw,ignoreeof TCP4:10.11.12.15:8899,ignoreeof'` Do i try to create a service for the socat command or can it be added to the aqualinkd service?

Also can I run the iAquaLink 3.0 device at the same time as the AquaLinkD? It seems to really bog down the system with both running.

**Comment by sjames99** [comment]
Anyone get this to work?  I can get it work nicely for about 5-10 minutes before the port gets overwhelmed and the serial connection dies and aqualink needs to restart.

EDIT: If I increase the buffer size it will run for about 20-30 minutes before it dies.

**Reply by sfeakes** [reply]
What do you mean by "port gets overwhelmed" ? 

There are a few people with this setup that works perfectly, and a few where it doesn't work reliably. When it doesn't work, it's always been due to the WiFi not being good enough.  Most/Many consumer grade WiFi hubs / setups are not going to have a good enough connection for this to work reliable.  WiFi Speed is not the issue, it's latency that will ALWAYS be the problem.  The fact that increasing the buffer allows it to work longer, sounds like a latency problem.

**Reply by sjames99** [reply]
It will go from bad packet to nothing to read to serial port blocked resetting.

**Reply by sfeakes** [reply]
That's absolutely latency.   The "local" port to AqualinkD is not getting "overwhelmed", there is nothing to be read on that port.  So either EW11 isn't sending it, SOCAT is buffering data, or more than lightly it's having WiFi issues.

UDP should work better than TCP for this type of thing, but I never got UDP to work on the EW11. (although I didn't spend that much time on UDP). You could try that, but ultimately I'd be looking at the WiFi and anything there you can do to optimize the connection.

**Reply by sjames99** [reply]
Funny you mention UDP I have been trying that this afternoon as an alternative but haven’t had any luck.

**Comment by bcre3306** [comment]
I concur with what is being said, I tried this for a month and it turned out being the WiFi latency. I would start with 20-30ms and it would jump to 200-400 after a while.

On Apr 11, 2025, at 6:11 PM, sfeakes ***@***.***> wrote:

﻿

That's absolutely latency. The "local" port to AqualinkD is not getting "overwhelmed", there is nothing to be read on that port. So either EW11 isn't sending it, SOCAT is buffering data, or more than lightly it's having WiFi issues.

UDP should work better than TCP for this type of thing, but I never for the EW11 to work on UDP. (although I didn't spend that much time on UDP). You could try that, but ultimately I'd be looking at the WiFi and anything there you can do to optimize the connection.

**Reply by sjames99** [reply]
I repeatedly pinged the EW11(Not from the pi) and only had one packet loss out of 500.  Minimum was 4ms, average was 23ms max was 380ms.  A few blips but no prolonged long responses.

I then brought up aqualinkd and let it run until I got the no serial connection error.  Once the error occurred I couldn't ping the EW11 any longer and it was down for about 5 minutes before it returned.

Interesting note, when the EW11 is operational, the old aqualink wifi connector was down.  Once the serial connection error happened, the aqualink wifi connector returned.

I just reviewed the logs with a fine tooth comb and noticed this:
Doesn't look like your USB2RS485 device (/dev/ttyEW11) supports low latency, this might cause problems on a busy RS485 bus (25): Inappropriate ioctl for device

Is this also part of my issue since the aqualink is so chatty?

**Reply by sfeakes** [reply]
the Jandy iAqualink wifi not working could be an AqualinkD setting.  if you have `enable_iaqualink=yes` in your aqualinkd config, that will stop the Jandy iAqualink from working since the panel can only support one device at that address.  With that turned off, AqualinkD will still work for all functionality it will just be a bit slower with setting virtual & one touch buttons.

The setting `ftdi_low_latency=YES` is for USB2RS485 devices. With this setting, AqualinkD will tell the Linux Kernel to disable any caching and set a high priority for the Serial interface, and a few other very hardware specific things for a FTDI serial chip.   You are not using a USB2RS485 adapter, so this setting is irrelevant.  You are "probably" using socat to create a virtual serial port (called EW11). It's the socat command that will set / handle any caching latency.

The ping isn't really a good test. but 23ms is REALLY pushing the boundaries of Jandy's RS485 interface and 380ms is way to slow.  The Jandy RS485 protocol works in the following manor.
Master device (the control panel) will sent a command to the RS485 network with a specific ID.
The slave device will read that command and has 20 to 40ms to reply.
If Master doesn't get the reply, it will move onto the next device/ID.
If the slave device replies too slowly, the master has already sent the next message, that can cause frame overlap / message collision and really created a problem for the Jandy control panel (it will do strange things like turning devices on/off randomly)

So in this you have the following. the message Control Panel->EW11->WIFI->SOCAT->AqualinkD. and the opposite direction for the reply. all has to be done in under 20 to 40ms.  If your wifi average is 23ms for a simply ping, that's a minimum 46ms for the master to receive the reply, simply for the wifi latency. That's not including any processing the EW11 has to do (twice, once for each direction), and AqualinkD.
AqualinkD at max will be 2ms even on very low powered hardware and super busy RS bus.

So that's probably why you are having such issues.  FYI, My average ping for EW11 is 2.5ms.

**Reply by sjames99** [reply]
I walked and map the WiFi strength around the pool and to where the aqua link is and it’s gets progressively worse as I get to that side of the house.  Great inside the wall that faces the aqua link.  I am going say something is interfering with signal.  I am going to move the EW11 away from that side and I have feeling it will improve drastically.  Thanks y’all for troubleshooting with me.  I’ll report back with my findings

**Reply by sjames99** [reply]
So I had a coupler and some cat5 cable left over.  I added the coupler and about 30ft of cat5 cable and moved the antenna towards the back middle of the house.  Ping times are a little better, but I have yet to get a failed TCP frame so that is improvement.  I also ran aqualinkd for 30 minutes and had zero dropped serial connections.  So progress!

**Reply by sfeakes** [reply]
Glad you are making progress.  Seems a lot of work when you could simply replace the EW11 with a Pi (or equivalent).  Out of interest, why are you not going that route.

**Reply by sjames99** [reply]
I had used a pi and usb adapter for the over two years but had to replace each of them multiple times as they died.  Not at the same time so was experimenting with alternatives as I couldn’t protect them from the elements or excessive heat no matter what I tried to protect them.  I don’t have space inside the AqualinkD box to store them so they had to be external.

**Reply by sfeakes** [reply]
When you get time, I’d be interested to see what / why / how the Pi’s failed.

I’ve been testing the Radxa Zero3 heavily recently, because it’s Pi Zero size but has eMMC (so no CF failures), and has better thermal protection. This with the AqualinkD-SerialHat & case, makes for a very small and more importantly stable device. IT’s only a touch bigger than EW11. The hat means it’s completely powered from the RS485 connections with no USB2RS485 or external power needed. AqualinkD talks to the RS485 through TTL ports on the GPIO.
This is not waterproof, but is small enough to fit in the low voltage side of the control panel.
Everything needed to build one is open source, and all information can be found in the repo’s listed www.aqualinkd.com.  (Site will have more information, it just lists the repo’s at moment). It’s still early phases of testing in external environments, but internal it’s far superior to Pi and USB2RS485 adapter.
BTW, The Radxa Zero3 is not as easy as the Pi to setup for this, but the aqualinkd-Radxa repo has a script you can run against their OS to make this easier. 

![IMG_0488](https://github.com/user-attachments/assets/11efda02-647e-4bb6-b79e-7976b6f030b2)
![IMG_0487](https://github.com/user-attachments/assets/e2952233-fe91-4f21-9c32-bac56678b8f1)
![IMG_0524](https://github.com/user-attachments/assets/935c3c86-dbb8-4a3b-b420-136bcf87c10d)

**Reply by sjames99** [reply]
Very cool!

first pi was a 3 worked for almost the entire year and then it died.  Had activity lights but wouldn’t boot just hang.  I think it got wet in a hurricane

2nd was a zero and it got too hot and burned up.  

3rd kept causing aqualink to power down after about 25 seconds.  Had to reset the breaker and remove the pi for it to power up.

two usb adapters failed.  First one went all goofy after about six months.  Just starting outputting garbage.  Would Get hot to the touch during this.

last one the usb portion died and couldn’t be recognized on any system.

So hence the experimenting with other options to see what is out there

EDIT: I might have just figure out why they kept dying.  My Aqualink panel is near my heat exchanger for the pool.  I noticed there is a decent amount of condensation around that area when the exchanger runs for a long a period of time.  I think it is the differences in the temperature and the air being expelled.  It wasn't the elements that was getting to these things, it was that.  I am going to try something different and try again with the pi and its placement and housing and I think it will be okay to survive.

**Comment by bcre3306** [comment]
I’m going to have to agree with Shawn.  I ran around with the EW11 for probably 2 months, finally gave up and went to the PI.  I wired it into a 24v -> 5v transformer from Amazon for like $7 and it worked great.  Been up and running without issues for months now.

From: sfeakes ***@***.***>
Date: Tuesday, April 15, 2025 at 8:47 AM
To: aqualinkd/AqualinkD ***@***.***>
Cc: bcre3306 ***@***.***>, Mention ***@***.***>
Subject: Re: [aqualinkd/AqualinkD] Wireless RS485 connection - EW11 (Discussion #336)

Glad you are making progress. Seems a lot of work when you could simply replace the EW11 with a Pi (or equivalent). Out of interest, why are you not going that route.

**Comment by jkosharek** [comment]
Are their instructions for the initial configuring AquaLinkD for the EW11? 
in the /etc/aqualinkd.conf I have:
serial_port=/dev/ttyEW11

In the docker version I see a aqexec-pre.sh and talk about this:
socat -d -d pty,link=/dev/ttyEW11,raw,ignoreeof TCP4:1.1.1.1:8899,ignoreeof &

If I run 'socat -d -d pty,link=/dev/ttyEW11,raw,ignoreeof TCP4:10.11.12.15:8899,ignoreeof' it's working but what is the best practice to get this to run as a service before AquaLinkD starts?

**Reply by bcre3306** [reply]
I would think the easiest would be to just create a service for aqexec-pre.sh or modify the aqualinkd service to call it.

**Reply by sfeakes** [reply]
As @bcre3306 mentioned. The best and most reliable way to do this would be to make a service that will start socat at boot.  There are a few things to also consider. AqualinkD will depend on this service being started prior to aqualinkd starting, so you would need to modify the aqualinkd service file to depend on the socat service file.  (ie if the system starts aqualinkd before socat, aqualinkd will fail). Creating a service is also the most work, and will require some knowledge of systemd services.

You could put socat in aqualinkd service file as something to execute before AqualinkD.  But now since it's constantly running you could end up with problems as one service file it starting (and trying to monitor) two executables.

You could put socat in cron to be executed 1 time at boot.  But if it fails it will never get restarted, and the process will never be monitored.

I HAVE NOT tested below, it's simply from memory so probably has some errors.

Create the following file.
```
# /etc/systemd/system/EW11-Bridge.service
[Unit]
Description=EW11 Bridge
After=network.target 
Requires=network.target

[Service]
Type=simple
StandardOutput=syslog
StandardError=syslog
SyslogIdentifier=EW11-Bridge

ExecStart=socat -d -d pty,link=/dev/ttyEW11,raw,ignoreeof TCP4:10.11.12.15:8899,ignoreeof
Restart=always

[Install]
WantedBy=multi-user.target
```

Modify the following
```
# /etc/systemd/system/aqualinkd.service
Change line
   Requires=network.target
to
  Requires=network.target EW11-Bridge.target
```

Then run `sudo systemctl daemon-reload` 

That should do it.  BUT please check for errors / typo's etc.  I could easily have something wrong.

**Reply by jkosharek** [reply]
Just figured this out and posted in another tread. Seems to be working well. Seems like a good addition to the Wiki

With AquaLinkD running should I be able to leave the iAquaLink 3.0 in place without any issues?

Might move this to a Home Assistant Add-On (Container) rather than a dedicated VM. This is brilliant though.

Do people switch from the scheduling in the control panel to something else or is there an GUI/interface I am missing in AquaLinkD for Schedules?

**Reply by sfeakes** [reply]
Yes you can leave iAqualink in place, as long as you have `enable_iaqualink=no` in aqualiinkd.conf.  If that's set to `yes`, then it'll stop iAqualink from working.

AquaklinkD scheduler information
https://github.com/aqualinkd/AqualinkD/wiki#Scheduler

I'm hesitant to put much information about EW11 in wiki since I don't want to deal with supporting it.  As you have found out there are tons of things that can go wrong that was way beyond the scope of AqualinkD.

**Reply by jkosharek** [reply]
Is it fine to have read_RS485_iAqualink = yes?

**Reply by sfeakes** [reply]
Yes, that's preferred.   `read_RS485_iAqualink` simply reads messages between the control panel and iAqualink, and will update AqualinkD status appropriately. `enable_iaqualink` will read and write messages to the iAqualink interface ID (ie the panel will think an iAqualink device is attached) and there for break the panel connection to the real iAqualink device.

**Reply by jkosharek** [reply]
I am seeing that and will move to a Pi 4 to play with. I just bought the Radxa Zero 3E 8GB. How do I buy the AqualinkD serial HAT and case or just have to print the case?

**Reply by sfeakes** [reply]
For both case and hat, You can use the repos to get someone to manufacture them for you.  or shoot me an email. sfeakes@gmail.com

-- EDIT --

On the Radxa,  Make sure you pick one with eMMC. and the cheapest available in all other respects.  So currently that's 1GB memory and 6GB eMMC.  The prices on amazon fluctuate a masive about, from $35 to $80 for the cheapest model

**Comment by dharp1935** [comment]
Working on a setup for a friend that doesn't have a hardwired connection to his pool controller. Was going to use the home assistant integration but the connection is very flakey. I've been using aqualinkd for a while and it is flawless. Want to thank the developer for this as it is very professional. Did some research and purchased a USR-W610 (serial to wifi). I was able to get it setup and communicating with my pool controller. Not sure if anyone has experience with this but happy to share the final setup and contribute to the group here as there are probably a lot of users that don't have a hardwired connection. Also, this device only pulls about 0.4A max and accepts 5 - 36VDC input. So don't see why it can't be connected to the aqualink serial power supply. Curious if anyone else has tried this. If not happy to share the setup information.

<img width="412" height="413" alt="image" src="https://github.com/user-attachments/assets/bb2cbcd0-e06f-4df4-9d76-ea4f97479cf0" />
